wzdx/standard_to_enhanced/planned_events_translator.py

```diff
@@ -0,0 +1,393 @@
+
+from wzdx.tools import date_tools, polygon_tools, wzdx_translator
+
+PROGRAM_NAME = 'PlannedEventsTranslator'
+PROGRAM_VERSION = '1.0'
+
+
+DEFAULT_PLANNED_EVENTS_FEED_INFO_ID = '49253be7-0c6a-4a65-8113-450f9041f989'
+
+
+def main():
+    input_file, output_file = parse_planned_events_arguments()
+    # Added encoding argument because of weird character at start of incidents.xml file
+
+    planned_events_obj = json.loads(open(input_file, 'r').read())
+    wzdx, event_type = wzdx_creator(planned_events_obj)
+    try:
+        event_type = wzdx.get('road_event_feed_info', wzdx['feed_info'])[
+            'features'][0]['properties']['core_details']['event_type']
+    except:
+        event_type = ''
+    schemas = {
+        'work-zone': wzdx_v40_feed.wzdx_v40_schema_string,
+        'restriction': road_restriction_v40_feed.road_restriction_v40_schema_string
+    }
+
+    if not wzdx_translator.validate_wzdx(wzdx, schemas[event_type]):
+        logging.error(
+            'validation error more message are printed above. output file is not created because the message failed validation.')
+        return
+    with open(output_file, 'w') as fwzdx:
+        fwzdx.write(json.dumps(wzdx, indent=2))
+        print('Your wzdx message was successfully generated and is located here: ' + str(output_file))
+
+
+# parse script command line arguments
+def parse_planned_events_arguments():
+    parser = argparse.ArgumentParser(
+        description='Translate Planned Event data to WZDx')
+    parser.add_argument('--version', action='version',
+                        version=f'{PROGRAM_NAME} {PROGRAM_VERSION}')
+    parser.add_argument('plannedEventsFile', help='planned_events file path')
+    parser.add_argument('--outputFile', required=False,
+                        default='planned_events_wzdx_translated_output_message.geojson', help='output file path')
+
+    args = parser.parse_args()
+    return args.plannedEventsFile, args.outputFile
+
+
+def wzdx_creator(message, info=None):
+    if not message:
+        return None
+    event_type = message['event']['type']
+
+   # verify info obj
+    if not info:
+        info = wzdx_translator.initialize_info(
+            DEFAULT_PLANNED_EVENTS_FEED_INFO_ID)
+    if not wzdx_translator.validate_info(info):
+        return None
+
+    if event_type == 'work-zone':
+        wzd = wzdx_translator.initialize_wzdx_object(info)
+        feature = parse_work_zone(message)
+    elif event_type == 'restriction':
+        wzd = wzdx_translator.initialize_wzdx_object_restriction(info)
+        feature = parse_road_restriction(message)
+    else:
+        logging.warning(f"Unrecognized event type: {message['event']['type']}")
+        return None
+
+    if feature:
+        wzd.get('features', []).append(feature)
+    if not wzd.get('features'):
+        return None
+
+    wzd = wzdx_translator.add_ids(wzd, event_type)
+    return wzd
+
+# {
```

Comment:
Do we need to keep this in here or can we reference it as a file?

Reply:
removed